core/blockchain/difficulty.go

```diff
@@ -302,6 +307,50 @@ func (b *BlockChain) calcNextRequiredDifficulty(block meerdag.IBlock, newBlockTi
 	return nextDiffBits, nil
 }
 
+// blockWindow returns a blockWindow of the given size that contains the
+// blocks in the past of startingNode, the sorting is unspecified.
+// If the number of blocks in the past of startingNode is less then windowSize,
+// the window will be padded by genesis blocks to achieve a size of windowSize.
+func (b *BlockChain) getBlockWindows(oldBlock meerdag.IBlock, powType pow.PowType, windowSize int) model.BlockWindow {
+	windows := make(model.BlockWindow, 0, windowSize)
+	count := 0
+	for i := uint64(0); ; i++ {
+		// Get the previous node while staying at the genesis block as
+		// needed.
+		if oldBlock == nil || !oldBlock.HasParents() {
+			break
+		}
+
+		for id := range oldBlock.GetParents().GetMap() {
```

Comment:
The `getBlockWindows` result of each run may be different because the selection of blocks is not detailed and specific enough